### PR TITLE
Redirect stats dashboard http to https

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Cleaned up the params a bit too.

### CODE REVIEW

File: puppet/modules/web/manifests/stats.pp [hunk 1]
```diff
@@ -19,13 +32,24 @@
...
   }
 
-  apache::vhost { 'shiny_server':
-    port          => '80',
-    servername    => $hostname,
-    docroot       => $webroot,
-    docroot_owner => $::apache::user,
-    docroot_group => $::apache::group,
-    proxy_pass    => $proxy_pass,
+  if $https_only {
```
Comment (ekohl) [inline puppet/modules/web/manifests/stats.pp]: Maybe we should unify this with `$https`? I don't see a reason to allow `https` but not run `https_only` these days.

Comment (GregSutcliffe) [reply puppet/modules/web/manifests/stats.pp]: Yeah, fair point. Fixed.

**Comment by GregSutcliffe** [comment]
I noticed Redmine is using this code without an exception for LE, and it seems to be working?

**Comment by ekohl** [comment]
That sounds correct. Looks like it prefers the actual files if they're present and we have the docroot set correctly.

File: puppet/modules/web/manifests/stats.pp [hunk 2]
```diff
@@ -19,16 +29,15 @@
...
-  }
-
   if $https {
+    apache::vhost { 'shiny-server':
+      port          => '80',
+      servername    => $hostname,
+      docroot       => $webroot,
+      docroot_owner => $::apache::user,
+      docroot_group => $::apache::group,
+      redirect_dest => "https://${servername}/",
```
Comment (ekohl) [inline puppet/modules/web/manifests/stats.pp]: 0710256ead6a01821ad5c0589e7eb19b49ddeee0 was needed to set this to `$hostname`

**Pull request merged**